semgrep-core/src/core/Rule.ml

```diff
@@ -40,6 +40,9 @@ type tok = AST_generic.tok [@@deriving show, eq, hash]
 
 type 'a wrap = 'a AST_generic.wrap [@@deriving show, eq, hash]
 
+(* To help report pattern errors in simple mode in the playground *)
```

Comment:
Perhaps clarify that the `string list` represents a path of pattern operators (correct?). You could also use a record here, usually preferable over tuples IMO.